### PR TITLE
Adopt new namespaced FairMQ API

### PR DESCRIPTION
**Pull request by dennisklein**
Because FairRoot is supporting at least the two latest FairSoft releases which still ship FairMQ v1.4, this PR proposes a solution that shall work with all FairMQ v1.4 releases.

### CODE REVIEW

**Review by ChristianTackeGSI — APPROVED** [review]
LGTM.

I like many of those changes. They make the stuff really easier to read / etc.

I added a few ideas.

**Review by rbx — APPROVED** [review]

File: base/MQ/FairRunFairMQDevice.h [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+#include <fairmq/Version.h>
+#ifdef FAIRMQ_VERSION_DEC
+#undef FAIRMQ_VERSION_DEC   // bugged
+#endif
+#if defined FAIRMQ_VERSION_MAJOR && defined FAIRMQ_VERSION_MINOR && defined FAIRMQ_VERSION_PATCH
+#define FAIRMQ_VERSION_DEC (FAIRMQ_VERSION_MAJOR * 100000) + (FAIRMQ_VERSION_MINOR * 1000) + (FAIRMQ_VERSION_PATCH * 10)
+#else
+#define FAIRMQ_VERSION_DEC 9999999
+#endif
```
[the remark this answers is not in the document]
Comment (dennisklein) [reply base/MQ/FairRunFairMQDevice.h:20]: oversight, thx. Added the sanity checks to the other header, too.

File: base/MQ/policies/Sampler/SimpleTreeReader.h [hunk 2]
```diff
@@ -19,10 +27,7 @@
 #include <TClonesArray.h>
 #include <TFile.h>
 #include <TTree.h>
-
-// FairRoot
-#include <FairMQLogger.h>
-#include <FairMQMessage.h>
+#include <fairlogger/Logger.h>
```
[the remark this answers is not in the document]
Comment (dennisklein) [reply base/MQ/policies/Sampler/SimpleTreeReader.h:30]: sure, will do

File: examples/MQ/pixelAlternative/src/devices/FairMQPixAltTaskProcessorBin.h [hunk 3]
```diff
@@ -101,16 +100,15 @@ class FairMQPixAltTaskProcessorBin : public FairMQDevice
             header->fRunId = payloadE->fRunId;
             header->fMCEntryNo = payloadE->fMCEntryNo;
             header->fPartNo = payloadE->fPartNo;
-            FairMQMessagePtr msgHeader(
-                NewMessage(header, sizeof(PixelPayload::EventHeader), [](void* data, void* /*hint*/) {
-                    delete static_cast<PixelPayload::EventHeader*>(data);
-                }));
+            auto msgHeader(NewMessage(header, sizeof(PixelPayload::EventHeader), [](void* data, void* /*hint*/) {
+                delete static_cast<PixelPayload::EventHeader*>(data);
+            }));
```
[the remark this answers is not in the document]
Comment (dennisklein) [reply examples/MQ/pixelAlternative/src/devices/FairMQPixAltTaskProcessorBin.h:105]: sure, will do

File: examples/advanced/Tutorial3/MQ/processorTask/FairTestDetectorMQRecoTaskTMessage.tpl [hunk 4]
```diff
@@ -1,3 +1,10 @@
+/********************************************************************************
+ * Copyright (C) 2014-2022 GSI Helmholtzzentrum fuer Schwerionenforschung GmbH  *
+ *                                                                              *
+ *              This software is distributed under the terms of the             *
+ *              GNU Lesser General Public Licence (LGPL) version 3,             *
+ *                  copied verbatim in the file "LICENSE"                       *
+ ********************************************************************************/
```
[the remark this answers is not in the document]
Comment (dennisklein) [reply examples/advanced/Tutorial3/MQ/processorTask/FairTestDetectorMQRecoTaskTMessage.tpl:7]: My general approach here is to treat the copyright fixes as some sort of explicit "access" time record (I do it manually, so no guarantees, that I overlook some cases) - but whenever I touch a file, I update its copyright year (or add a missing copyright). So far, this belonged together with the change.

(I always wanted to write a vim plugin that auto-manages the copyright lines on save, but never found the time for it)

[the remark this answers is not in the document]
Comment (dennisklein) [reply examples/advanced/Tutorial3/MQ/processorTask/FairTestDetectorMQRecoTaskTMessage.tpl:7]: Some files in this PR contain only the added copyright header though, for those I will make a separate commit!

[the remark this answers is not in the document]
Comment (ChristianTackeGSI) [reply examples/advanced/Tutorial3/MQ/processorTask/FairTestDetectorMQRecoTaskTMessage.tpl:7]: Sounds good!

**Review by ChristianTackeGSI — APPROVED** [review]

**Comment by dennisklein** [comment]
The [failed test](https://cdash.gsi.de/testDetails.php?test=11410547&build=325682) looks like a false positive (log output is fine). Merging.

**Pull request merged**